gtars/src/uniwig/writing.rs

```diff
@@ -115,3 +120,91 @@ pub fn write_to_wig_file(
     }
     buf.flush().unwrap();
 }
+
+pub fn write_to_bed_graph_file(
+    counts: &[u32],
+    filename: String,
+    chromname: String,
+    start_position: i32,
+    stepsize: i32,
+) {
+    let path = std::path::Path::new(&filename).parent().unwrap();
+    let _ = create_dir_all(path);
+    let mut position = start_position;
+
+    let file = OpenOptions::new()
+        .create(true) // Create the file if it doesn't exist
+        .append(true) // Append data to the existing file if it does exist
+        .open(filename)
+        .unwrap();
+
+    let mut buf = BufWriter::new(file);
+
+    for count in counts.iter() {
+        writeln!(
+            &mut buf,
+            "{}\t{}\t{}\t{}",
+            chromname,
+            position,
+            position + stepsize,
+            count
+        )
+        .unwrap();
+        position = position + stepsize;
+    }
+    buf.flush().unwrap();
+}
+
+/// Converts uniwig generated bedGraphs to bigWig files
+pub fn write_bw_files(location: &str, chrom_sizes: &str, num_threads: i32, zoom_level: i32) {
```

Comment:
What about changing the type of `num_threads` and `zoom_level` to `u32`? They are being converted anyways down there